### PR TITLE
[SDP-1189] Setup Circle SDK

### PR DESCRIPTION
**Pull request by marwen-abid**
### What
* Setting up the Circle SDK 
* Adding `GET /ping` call https://developers.circle.com/circle-mint/reference/ping
* Adding `POST /transfers` call https://developers.circle.com/circle-mint/reference/createbusinesstransfer
* Adding `GET /transfers/:id` call https://developers.circle.com/circle-mint/reference/getbusinesstransfer

### Why
https://stellarorg.atlassian.net/browse/SDP-1189

#### PR Structure

* [x] This PR has a reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR title and description are clear enough for anyone to review it.
* [x] This PR does not mix refactoring changes with feature changes (split into two PRs otherwise).

#### Thoroughness

* [x] This PR adds tests for the new functionality or fixes.
* [x] This PR contains the link to the Jira ticket it addresses.

#### Configs and Secrets

* [x] No new CONFIG variables are required -OR- the new required ones were added to the helmchart's [`values.yaml`] file.
* [x] No new CONFIG variables are required -OR- the new required ones were added to the deployments ([`pr-preview`], [`dev`], [`demo`], `prd`).
* [x] No new SECRETS variables are required -OR- the new required ones were mentioned in the helmchart's [`values.yaml`] file.
* [x] No new SECRETS variables are required -OR- the new required ones were added to the deployments ([`pr-preview secrets`], [`dev secrets`], [`demo secrets`], `prd secrets`).

#### Release

* [x] This is not a breaking change.
* [x] **This is ready for production.**. If your PR is not ready for production, please consider opening additional complementary PRs using this one as the base. Only merge this into `develop` or `main` after it's ready for production!

#### Deployment

* [ ] Does the deployment work after merging?

[`values.yaml`]: ../helmchart/sdp/values.yaml
[`pr-preview`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/stellar-disbursement-platform/backend-helm-values
[`dev`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/backend-helm-values
[`demo`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-backend-helm-values
[`pr-preview secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/externalsecrets-common-previews.yaml#L241-L346
[`dev secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/stellar-disbursement-platform-externalsecrets.yaml
[`demo secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-sdp-externalsecrets.yaml

### CODE REVIEW

File: internal/circle/errors.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+// parseAPIError parses the error response from Circle APIs.
+// https://developers.circle.com/circle-mint/docs/circle-apis-api-errors.
+func parseAPIError(resp *http.Response) (*APIError, error) {
+	if resp.StatusCode == http.StatusUnauthorized {
+		return nil, fmt.Errorf("unauthorized to access the resource")
+	}
+
+	if resp.StatusCode == http.StatusForbidden {
+		return nil, fmt.Errorf("forbidden to access the resource")
+	}
```
Comment (marcelosalloum) [inline internal/circle/errors.go:40]: 👋  I think these responses should trigger an API error, so the caller can identify the 401/403 issue, and decide what to do with it. WDYT?

**Review by marcelosalloum — APPROVED** [review]
LGTM! Feel free to merge it after addressing the comments below. I marked the most important one with the 👋 emoji.

File: internal/circle/errors.go (hunk 1 above)
Comment (marwen-abid) [reply internal/circle/errors.go:40]: So `APIError` here is a Circle resource that's returned (mostly with a 400 http status) and that signifies there is a user error with the API. The code in the APIError is not the http status code, but a Circle specific code as described here https://developers.circle.com/circle-mint/docs/circle-apis-api-errors 

Given that 401 and 403 issues prevent us from even fetching the APIError, I handled them differently. I'll try to think of a way to propagate these properly.

Comment (marcelosalloum) [reply internal/circle/errors.go:40]: Hum, when I executed the request with an invalid key I did get the API error with the 401 status code:

```curl
curl --location 'https://api-sandbox.circle.com/v1/transfers' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer aaa' \
--write-out '\nHTTP Status: %{http_code}\n------\n'
```

Returned:
```text
{"code":401,"message":"Malformed key. Does it contain three parts?"}
HTTP Status: 401
```

File: internal/circle/client.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+	if pingResp.Message == "pong" {
+		return true, nil
+	}
+
+	return false, fmt.Errorf("unexpected response message: %s", pingResp.Message)
+}
+
+// PostTransfer creates a new transfer.
+// https://developers.circle.com/circle-mint/reference/createbusinesstransfer.
```
[the remark this answers is not in the document]
Comment (marwen-abid) [reply internal/circle/client.go]: Addressed in 7d3e7fc1545096dec08b5c18045d787f46f4e3f9

File: internal/circle/client.go [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+	}
+
+	return parseTransferResponse(resp)
+}
+
+// request makes an HTTP request to the Circle API.
+func (client *Client) request(ctx context.Context, u string, method string, isAuthed bool, body io.Reader) (*http.Response, error) {
+	req, err := http.NewRequestWithContext(ctx, method, u, body)
+	if err != nil {
+		return nil, err
```
[the remark this answers is not in the document]
Comment (marwen-abid) [reply internal/circle/client.go]: Addressed in 7d3e7fc1545096dec08b5c18045d787f46f4e3f9

File: internal/circle/client_test.go [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+	httpclientMocks "github.com/stellar/stellar-disbursement-platform-backend/internal/serve/httpclient/mocks"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+func Test_NewClient(t *testing.T) {
+	cc := NewClient(Production, "test-key")
+	assert.Equal(t, "https://api.circle.com", cc.BasePath)
+	assert.Equal(t, "test-key", cc.APIKey)
+}
```
[the remark this answers is not in the document]
Comment (marwen-abid) [reply internal/circle/client_test.go]: Addressed in 7d3e7fc1545096dec08b5c18045d787f46f4e3f9

File: internal/circle/client_test.go [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+
+func Test_Client_GetTransferByID(t *testing.T) {
+	ctx := context.Background()
+	t.Run("get transfer by id error", func(t *testing.T) {
+		httpClientMock := httpclientMocks.NewHttpClientMock(t)
+		cc := &Client{
+			BasePath:   "http://localhost:8080",
+			APIKey:     "test-key",
+			httpClient: httpClientMock,
+		}
```
[the remark this answers is not in the document]
Comment (marwen-abid) [reply internal/circle/client_test.go]: Addressed in 7d3e7fc1545096dec08b5c18045d787f46f4e3f9

**Pull request merged**